src/main/java/org/opensearch/neuralsearch/search/util/HybridSearchResultFormatUtil.java

```diff
@@ -70,6 +70,30 @@ public static FieldDoc createFieldDocDelimiterElementForHybridSearchResults(fina
         return new FieldDoc(docId, MAGIC_NUMBER_DELIMITER, fields);
     }
 
+    /**
+     * Checking if passed scoreDocs object is a special element (start/stop or delimiter) in the list of hybrid query result scores
+     * @param scoreDoc score doc object to check on
+     * @return true if it is a special element
+     */
+    public static boolean isHybridQuerySpecialElement(final ScoreDoc scoreDoc) {
```

Comment:
Do not review this class as it already merged in main. The change is showing here because we have cherry picked the commit.